docs/deployment_kubernetes_docker.md

```diff
@@ -0,0 +1,199 @@
+    * Installation instruction: https://www.docker.com/get-started
+3. Python (3.6 or above) and required packages: `bentoml`, `fastai`
+    * ```pip install bentoml fastai```
+
+
+## Save pet classification model with BentoML
+
+Follow the notebook from Fastai lesson one to train a pet classification model.
+
+An example base on that notebook is the following:
+
+```python
+from fastai.vision import *
+
+path = untar_data(URLs.PETS)
+path_img = path/'images'
+fnames = get_image_files(path_img)
+pat = re.compile(r'/([^/]+)_\d+.jpg$')
+bs=64
+data = ImageDataBunch.from_name_re(path_img, fnames, pat, ds_tfms=get_transforms(),
+                                   size=299, bs=bs//2).normalize(imagenet_stats)
+learn = create_cnn(data, models.resnet50, metrics=error_rate)
+learn.fit_one_cycle(8)
+learn.unfreeze()
+learn.fit_one_cycle(3, max_lr=slice(1e-6,1e-4))
+```
+
+### Define API server with BentoML
+
+Save the following code to a file named `pet_classification.py`:
+
+```python
+from bentoml import BentoService, api, env, artifacts
+from bentoml.artifact import FastaiModelArtifact
+from bentoml.handlers import FastaiImageHandler
+
+@artifacts([FastaiModelArtifact('pet_classifier')])
+@env((auto_pip_dependencies=True)
+class PetClassification(BentoService):
+
+    @api(FastaiImageHandler)
+    def predict(self, image):
+        result = self.artifacts.pet_classifier.predict(image)
+        return str(result)
+```
+
+This code defines a prediction server using `Fastai` model, asks BentoML to figure out
+the required PyPi packages automatically. It also defined an API called `predict`, that
+is the entry point to access this prediction service. The API is expecting a `Fastai`
+`ImageData` object as its input data.
+
+Run the following code to create a BentoService SavedBundle with the pet classification
+model. It's a versioned file archive ready for production deployment.  The archive
+contains the prediction service defined above, python code dependencies and PyPi
+dependencies, and the trained pet classification model. :
+
+```python
+# 1) import the custom BentoService defined above
+from pet_classification import PetClassification
+
+# 2) `pack` it with required artifacts
+service = PetClassification()
+service.pack('pet_classifier', learn)
+
+# 3) save your BentoService
+service.save()
+```
+
+### Validate the saved bundle by using prediction request with sample data
+
+BentoML automatically process the incoming data into required data format defined in the
+API. For the pet classifier BentoService defined above, incoming data will transform to
+fastai `ImageData` object.
+
+Make prediction request with sample data using BentoML CLI:
+
+```bash
+# Replace PATH_TO_TEST_IMAGE_FILE with one of the image from {path_img}
+# An example path: /Users/user_name/.fastai/data/oxford-iiit-pet/images/shiba_inu_122.jpg
+
+bentoml run PetClassification:latest predict --input=PATH_TO_TEST_IMAGE_FILE
+```
```

Comment:
I think we should probably use `bentoml serve` here to test sending requests. mostly people won't think of CI/CD testing when trying out a new deployment solution...  here they just need to know I can start a REST API server